service/mantle/shopify/ApiServices.xml

```diff
@@ -1452,57 +1452,60 @@
-            </entity-find>
+            <!-- only shopify order are processed -->
+            <if condition="order.salesChannelEnumId == 'ScShopify'">
+                <set field="refund_line_items" from="[]"/>
+                <set field="refund" from="[:]"/>
+                <set field="shipping" from="[:]"/>
+                <entity-find entity-name="mantle.order.OrderItem" list="orderItems">
+                    <econdition field-name="orderId"  from="order.orderId"/>
+                    <econdition field-name="itemTypeEnumId"  value="ItemProduct"/>
+                    <econdition field-name="quantityCancelled"   operator="greater" from="0.0"/>
+                </entity-find>
 
-            <iterate list="orderItems" entry="item">
-                <script>
-                    def items = [:]
-                    items.line_item_id = item.externalItemSeqId
-                    items.quantity = item.quantityCancelled
-                    items.restock_type = 'cancel'
-                    refund_line_items.add(items)
-                </script>
-            </iterate>
+                <iterate list="orderItems" entry="item">
+                    <script>
+                        def items = [:]
+                        items.line_item_id = item.externalItemSeqId
+                        items.quantity = item.quantityCancelled
+                        items.restock_type = 'cancel'
+                        refund_line_items.add(items)
+                    </script>
+                </iterate>
 
-            <set field="externalId" from="order.externalId" />
-            <set field="currencyUomId" from="order.currencyUomId" />
-            <set field="shipping.full_refund" value="true"/>
-            <set field="refund.shipping" from="shipping"/>
-            <set field="refund.currency" from="currencyUomId"/>
-            <set field="refund.refund_line_items" from="refund_line_items"/>
-            <set field="refund.orderStatus" from="order.statusId" />
-            <set field="refund.orderId" from="externalId" />
-            <set field="messageText" from="groovy.json.JsonOutput.toJson([refund:refund])"/>
+                <set field="externalId" from="order.externalId" />
+                <set field="currencyUomId" from="order.currencyUomId" />
+                <set field="shipping.full_refund" value="true"/>
+                <set field="refund.shipping" from="shipping"/>
+                <set field="refund.currency" from="currencyUomId"/>
+                <set field="refund.refund_line_items" from="refund_line_items"/>
+                <set field="refund.orderStatus" from="order.statusId" />
+                <set field="refund.orderId" from="externalId" />
+                <set field="messageText" from="groovy.json.JsonOutput.toJson([refund:refund])"/>
+
+                <!-- The service to canculate refund order is called only if exist items to cancel -->

```

Comment:
<!-- The service to calculate refund order is called only if exist items to cancel -->

Reply:
Done